api/include/opentelemetry/context/runtime_context.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "opentelemetry/context/context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+// Provides a wrapper for propagating the context object globally. In order
+// to use either the threadlocal_context.h file must be included or another
+// implementation which must be derived from the RuntimeContext can be
+// provided.
+class RuntimeContext
+{
+public:
+  class Token
+  {
+  public:
+    bool operator==(const Context &other) { return (*context_ == other); }
+
+    // A constructor that sets the token's Context object to the
+    // one that was passed in.
+    Token(Context *context) { context_ = context; }
+
+    Token() = default;
+
+  private:
+    friend class RuntimeContext;
+
+    Context *context_;
+
+    // Returns the stored context object.
+    Context *GetContext() { return context_; }
+  };
+
+  // Return the current context.
+  static Context *GetCurrent() { return context_handler_->InternalGetCurrent(); }
+
+  // Dummy method to be overriden by derived class implementation
+  virtual Context *InternalGetCurrent() { return nullptr; }
```

Comment:
The things that the implementation overrides should be private (or protected if necessary).

Reply:
Made protected.